src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -470,6 +481,41 @@ MethodSymbol getTopLevelMethod(MethodSymbol method)
                 }
                 return null;
             }
+
+            bool isSupportedConversion(Conversion conversion, BoundExpression operandOpt)
+            {
+                // https://github.com/dotnet/roslyn/issues/32599: Allow implicit and explicit
+                // conversions where the nullable state of the operand remains valid.
+                switch (conversion.Kind)
+                {
+                    case ConversionKind.Identity:
+                    case ConversionKind.DefaultOrNullLiteral:
+                        return true;
+                    case ConversionKind.ImplicitReference:
+                        return operandOpt?.IsLiteralNullOrDefault() == true;
```

Comment:
Not sure I understand this? Literals won't have a slot, so why do we want to support literal conversion?

Reply:
The literal `null` or `default` may be nested in a tuple, `(null, x)` for instance, and we should generate a slot for the tuple.

---
In reply to: [249076864](https://github.com/dotnet/roslyn/pull/32612#discussion_r249076864) [](ancestors = 249076864)